docs/natspec-format.rst

```diff
@@ -0,0 +1,180 @@
+##############
+NatSpec Format
+##############
+
+Solidity contracts can use a special form of comments to provide rich
+documentation for functions, return variables and more. This special form is
+named the Ethereum Natural Language Specification Format (NatSpec).
+
+This documentation is segmented into developer-focused messages and end-user-
```

Comment:
Will the newline create a space in the rendered document?

Reply:
This is currently undefined behavior as far as how I have written the documentation. But the actual implementation is Solidity is that newlines in the docblock are concatenated with the newline deleted. A full demonstration follows.

Our options at this point are:

1. Leave the issue of whitespace as undefined behavior for #6122 <kbd>RECOMMENDED</kbd>
2. Add whitespace handling to the specification

Please advise.

# Test case

```solidity
pragma solidity ^0.5.6;

/// @title A simulator for trees
/// extended title
/// @author Larry A. Gardner
///  extended author
/// @notice You can use this contract for only the most basic simulation
///         extended notice
/// @dev All function calls are currently implemented without side effects
/// @dev Repeated dev
contract Tree {
    /// @author Mary A. Botanist
    /// @notice Calculate tree age in years, rounded up, for live trees
    /// @dev The Alexandr N. Tetearing algorithm could increase precision
    /// @param rings The number of rings from dendrochronological sample
    /// @return age in years, rounded up for partial years
    function age(uint256 rings) external pure returns (uint256) {
        return rings + 1;
    }
}
```

Rendered output:

```json
{
  "author" : "Larry A. Gardner  extended author",
  "details" : "All function calls are currently implemented without side effects Repeated dev",
  "methods" : 
  {
    "age(uint256)" : 
    {
      "author" : "Mary A. Botanist",
      "details" : "The Alexandr N. Tetearing algorithm could increase precision",
      "params" : 
      {
        "rings" : "The number of rings from dendrochronological sample"
      },
      "return" : "age in years, rounded up for partial years"
    }
  },
  "title" : "A simulator for trees extended title"
}
```

and

```
{
  "methods" : 
  {
    "age(uint256)" : 
    {
      "notice" : "Calculate tree age in years, rounded up, for live trees"
    }
  },
  "notice" : "You can use this contract for only the most basic simulation         extended notice"
}
```

Reply:
I think this was a misunderstanding. I was talking about the rst rendering, not about natspec.